src/main/kotlin/com/michaelsgroi/baseballreference/Career.kt

```diff
@@ -18,8 +24,51 @@ data class Career(
     val playerName: String,
     val war: Double,
     val seasonLines: List<SeasonLine>,
+    val awardVoteLines: List<RankedAwardShare>,
     val warPercentile: Double = 0.0,
 ) {
+    private val seasonLinesBySeason: Map<Int, List<SeasonLine>> by lazy { seasonLines.groupBy { it.season() } }
+    private val awardVotesBySeason: Map<Int, List<RankedAwardShare>> by lazy { awardVoteLines.groupBy { it.season } }
+
+    init {
+        validate()
+    }
+
+    private fun validate() {
+        // player id's
+        val seasonLinePlayerIds = seasonLines.map { it.playerId() }.toSet()
+        val awardLinePlayerIds = awardVoteLines.map { it.playerId }.toSet()
+        require(
+            seasonLinePlayerIds.size == 1
+                    && (awardLinePlayerIds.isEmpty() || awardLinePlayerIds.size == 1)
+                    && seasonLinePlayerIds.first() == playerId
+                    && (awardLinePlayerIds.isEmpty() || awardLinePlayerIds.first() == playerId)
+        ) {
+            "seasonLinePlayerIds($seasonLinePlayerIds) and " +
+                    "awardLinePlayerIds($awardLinePlayerIds) must match parent playerId=$playerId"
+        }
+
+        // seasons
+        val seasonLineSeasons = seasonLinesBySeason.keys
+        val awardVoteSeasons = awardVotesBySeason.keys
+        require(seasonLineSeasons.containsAll(awardVoteSeasons))
+        { "award vote season($awardVoteSeasons) contains a season not found in season line seasons($seasonLineSeasons)" }
+
+        // leagues
+        val seasonLineLeagues = seasonLines.map { it.league() }.toSet()
+        val awardLineLeagues = awardVoteLines.map { it.league }.toSet().minus("ML")
```

Comment:
consider ` - "ML"` instead of `.minus("ML")`

BTW: why remove `"ML"`?

Reply:
The reason for removing is that I'm comparing the "leagues"(AL vs NL) the player has played in according to their "season lines" against the league associated with the awards that they received votes for.  There are awards that(at one time, maybe still today, I didn't look at the data to find out) are issued the major league-level.  So it's expected that the set of unique leagues referenced in the award voting may include `ML` and distinct set of leagues pulled from the season lines would not.

Reply:
makes sense. thanks.

(consider an enum for these (e.g., w/instance `MAJOR_LEAGUE("ML")`)?)

Reply:
Yes, I put in a TODO to add an enum here.